web/src/main/java/com/futurewei/alcor/web/restclient/IpManagerRestClient.java

```diff
@@ -84,11 +84,18 @@ public IpAddrRequest allocateIpAddress(IpVersion ipVersion, String vpcId, String
     @DurationStatistics
     public IpAddrRequest allocateIpAddress(IpAddrRequest ipAddrRequest) throws Exception {
         HttpEntity<IpAddrRequest> request = new HttpEntity<>(ipAddrRequest);
-        IpAddrRequest result = restTemplate.postForObject(ipManagerUrl, request, IpAddrRequest.class);
-
-        verifyAllocatedIpAddr(result);
-
-        return result;
+        int count = 0;
+        int maxTries = 3;
+        while(true) {
+            try {
+                IpAddrRequest result = restTemplate.postForObject(ipManagerUrl, request, IpAddrRequest.class);
+                verifyAllocatedIpAddr(result);
+                return result;
+            } catch (Exception e) {
+                // handle exception
+                if (++count == maxTries) throw e;
+            }
+        }
```

Comment:
@DavidLiu506 Should we use formal way to do the retry? such as `RetryTemplate` described in this link: https://www.baeldung.com/spring-retry

Reply:
Nice. Thank you!

Reply:
@cj-chung fixed using spring framework.